### PR TITLE
pkp/pkp-lib#5122 Facilitate the use of Iterators in service classes

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: classes/search/ArticleSearchIndex.inc.php [hunk 1]
```diff
@@ -277,11 +271,11 @@ public function rebuildIndex($log = false, $journal = null, $switches = array())
...
-				$articles = $submissionDao->getByContextId($journal->getId());
-				while ($article = $articles->next()) {
-					if ($article->getSubmissionProgress() == 0) { // Not incomplete
-						$this->submissionMetadataChanged($article);
-						$this->submissionFilesChanged($article);
+				$submissions = Services::get('submission')->getMany(['contextId' => $journal->getId()]);
+				foreach ($submissions as $submission) {
+					if ($submission->getSubmissionProgress() == 0) { // Not incomplete
+						$this->submissionMetadataChanged($submission);
+						$this->submissionFilesChanged($submission);
```
Comment (NateWr) [inline classes/search/ArticleSearchIndex.inc.php]: Not sure, but should this only include published submissions? If so, you can add a `status` property to the `getMany` call:

```
Services::get('submission')->getMany(['contextId' => $journal->getId(), 'status' => STATUS_PUBLISHED]);
```

File: controllers/grid/articleGalleys/form/ArticleGalleyForm.inc.php [hunk 2]
```diff
@@ -60,23 +60,23 @@ function($galleyLocale) use ($journal) {
...
-	function fetch($request) {
-		$journal = $request->getJournal();
+	function fetch($request, $template = null, $display = false) {
 		$templateMgr = TemplateManager::getManager($request);
 		if ($this->_articleGalley) $templateMgr->assign(array(
 			'representationId' => $this->_articleGalley->getId(),
 			'articleGalley' => $this->_articleGalley,
 			'articleGalleyFile' => $this->_articleGalley->getFile(),
 		));
+		$journal = $request->getJournal();
```
Comment (NateWr) [inline controllers/grid/articleGalleys/form/ArticleGalleyForm.inc.php]: `$context = $request->getContext();` :)

File: package.json [hunk 3]
```diff
@@ -13,29 +13,29 @@
...
+    "babel-eslint": "^10.0.3",
     "element-resize-event": "^3.0.3",
     "eslint": "^5.16.0",
-    "eslint-plugin-vue": "^5.2.2",
-    "less": "^3.0.4",
+    "eslint-plugin-vue": "^5.2.3",
+    "less": "^3.10.3",
     "less-loader": "^4.1.0",
-    "lint-staged": "^8.1.6",
+    "lint-staged": "^8.2.1",
```
Comment (NateWr) [inline package.json]: There are a lot of version bumps here. Usually when I run an update like this, I do some manual checking to try to identify any regressions... We also need to keep the versions here in sync with the version in the UI Library.

So I'd suggest either not including these changes in a commit (I'm happy to run an update whenever if you'd like) or making sure that the same changes happen in UI Library and doing some testing on it...

File: classes/search/ArticleSearchIndex.inc.php (hunk 1 above)
Comment (asmecher) [reply classes/search/ArticleSearchIndex.inc.php]: Depends on our indexing strategy -- there's value in indexing unpublished content if we don't re-index fully on publication, which may be the case. This needs an audit/clean-up -- I'll do a small assessment as part of this issue and invariably leave some for https://github.com/pkp/pkp-lib/issues/3180 as well.

File: package.json (hunk 3 above)
Comment (asmecher) [reply package.json]: OK, happy to remove these!

File: controllers/grid/articleGalleys/form/ArticleGalleyForm.inc.php (hunk 2 above)
Comment (asmecher) [reply controllers/grid/articleGalleys/form/ArticleGalleyForm.inc.php]: Just juggling already-existing code, but yes, fixed!

**Pull request merged**